### DISCUSSION TITLE
v5.1.0

### DISCUSSION THREAD

**Discussion by richardwilkes** [richardwilkes/gcs#570] [Announcements]
### New Features

- You can now drag content to anywhere on a sheet or template and if it isn't over an actual drop target, an appropriate drop target is chosen and highlighted for you (#557).
- Double-clicking on a folder or library in the Library Explorer now toggles its open state.
- The amount to multiply the scroll wheel movement by is now displayed and editable by users in the General Settings view (#558).
- Added an additional color for the interior divider lines (#550).
- Rearranged the controls in some of the editor views to make a field the first item, allowing it to be focused upon creation (part of #567).
- Command/Control-Enter now acts like pressing the commit button in editor views (part of #567).
- Escape now acts like pressing the cancel button in editor views (part of #567).
- Various paths GCS uses are now displayed in the General Settings view as well as in the command line help output.

### Bugs Fixed

- Fixed tooltips not showing properly for the selected options on the sheet (#569).
- Fixed the ability to delete attributes from the sheet (#562).
- Fixed the file dialog failing to remember where it was last used (#536). This may still not work reliably on some platforms. Please report any additional issues you find around this.
- Fixed many cases of the keyboard focus jumping around inappropriately. I'm sure there are more left -- please report them as you find them.
- Fixed PDF generation on machines that have a monitor which does not report valid EDID data (#553). Previously, depending on the bad values, it could try to scale up the PDF so large that it could not allocate enough space for it in memory.
- Fixed disabled trait containers so that they and their contents no longer count towards the point total (#551).
- Fixed infinite loop when trying to paginate the display for printing or PDF generation. This was caused by excessively large row content that was too large for a single row to fit on a logical page. The code now guards against this, but does not yet attempt to try to break the large row up into smaller chunks. Instead, you just don't get to see the portion that goes off of the page.
- Fixed equipment added to the carried section of a sheet via the menus not automatically enabling the equipped flag (#556).
- Fixed the option to not include unspent points in the overall point total (#547). This is now a per-sheet setting rather than a global setting.
- Fixed string matching when lists were involved (such as for tags). Multiple entries in the list only worked previously when all entries in the list matched, which is only correct for about half of the possible matcher types. The other half only need at least one match.
- Fixed some background errors that would happen when trying to write the GCS document icons on Windows.
- Fixed an inability to open the about box more than once per session.

### Library Compatibility
- Loads library file formats v3.x.x and v4.x.x. Only produces v4.x.x. Note that GCS v4.30.0 is the oldest version that produced the v3.x.x library file format. You will need to load and save your data with a version capable of writing the v3.x.x format before you will be able to load it into this release.

<hr /><em>This discussion was created from the release <a href='https://github.com/richardwilkes/gcs/releases/tag/v5.1.0'>v5.1.0</a>.</em>

**Comment by jutis** [comment]
What happened to the debian package? Now its just a gcs binary in the linux distribution

**Reply by richardwilkes** [reply]
Correct. This is the simplest possible method of distribution. GCS is perfectly happy to run from wherever you place it and doesn't require anything more.

**Comment by jutis** [comment]
Hi

Well you might think so but since there are no dependency checks done 
with a straight binary and the binary does not include the libraries 
needed, it will not run unless those libraries are installed

A packetmanager installation would rectify this

br
/jimmy

On 2022-10-02 18:54, Richard Wilkes wrote:
>
> Correct. This is the simplest possible method of distribution. GCS is 
> perfectly happy to run from wherever you place it and doesn't require 
> anything more.
>
> —
> Reply to this email directly, view it on GitHub 
> <https://checkpoint.url-protection.com/v1/url?o=https%3A//github.com/richardwilkes/gcs/discussions/570%23discussioncomment-3782415&g=OWIyOTgyNjEzODMyY2ZjOQ==&h=ZmY3ZDU3N2QzNTI0NWEwNjg5NjU4N2NkM2JiZTQ5OWVmNGIyY2NiYjY2OTY3ZDcwM2JkNjc1YmRiOTgzMWFkMQ==&p=Y3AxOnNvdXJjZWNvbTpjOm86NTUwNzIyODM1OTYyOWFjMTFhMDcwZjQwZmM5NzU1YTc6djE6aDpU>, 
> or unsubscribe 
> <https://checkpoint.url-protection.com/v1/url?o=https%3A//github.com/notifications/unsubscribe-auth/AD4L5I3SIJDBWCKSME4YWJDWBG43BANCNFSM6AAAAAAQ2HZHVM&g=MjY4ODAxOTY2NGY3MGE5Yg==&h=NjljYzI5MjM2MWYwMmNlZWJjNWI4ZmNlZjdlZjFhZjc3YjI2ZjFlNDVmZTcxOTFiMGVjNjUwNGQ2NjI3Y2NlOQ==&p=Y3AxOnNvdXJjZWNvbTpjOm86NTUwNzIyODM1OTYyOWFjMTFhMDcwZjQwZmM5NzU1YTc6djE6aDpU>.
> You are receiving this because you commented.Message ID: 
> ***@***.***>
>

-- 
Jimmy Jutwreten              Sourcecom Svenska AB
Senior Security Consultant   c/o iOFFICE Kungsgatan 64
Mobile +46 (0)705 683400     111 22 Stockholm
***@***.***           https://www.sourcecom.se

**Comment by richardwilkes** [comment]
The binary does include everything it possibly can. There are certain shared libraries that each platform refuses to work properly with if embedded statically which are not under my control. A package manager does not change this.

Anyway, I have zero plans to go back to installers on any platform, as they just created headaches for me. For Linux, as of GCS 5.2, I've moved the build machine back to a much older release of Linux to build it on, which means it will be compatible with more versions of Linux. If you are using Linux and haven't tried that one yet, you should to see if it will now run for you.
